src/sortStudents.ts

```diff
@@ -1,16 +1,74 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  type Sum = (a: number, b: number) => number;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const sortedStudents: Student[] = [...students];
+  const sum: Sum = (a, b) => a + b;
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        sortedStudents.sort((a: Student, b: Student) => a[sortBy]
+          .localeCompare(b[sortBy]));
+      }
+      break;
+    case SortType.Surname:
+      if (order === 'asc') {
+        sortedStudents.sort((a: Student, b: Student) => a[sortBy]
+          .localeCompare(b[sortBy]));
+      }
+      break;
+    case SortType.Age:
+      if (order === 'asc') {
+        sortedStudents
+          .sort((a: Student, b: Student) => +a[sortBy] - +b[sortBy]);
+      } else {
+        sortedStudents
+          .sort((a: Student, b: Student) => +b[sortBy] - +a[sortBy]);
```

Comment:
```suggestion
          .sort((a: Student, b: Student) => +a[sortBy] - +b[sortBy]);
      } else {
        sortedStudents
          .sort((a: Student, b: Student) => Number(b[sortBy]) - Number(a[sortBy]));
```

Это необязательно, но мне кажется, что так код становится читабельнее:)